### PR TITLE
Mobs lighting

### PR DESCRIPTION
**Pull request by grit96**
Only spawn mobs in darkness.
This will need to be reworked to create the lighting buffer separately from the render process, and to support the Python renderer.

#136

### CODE REVIEW

File: mobs.py [hunk 1]
```diff
@@ -90,9 +90,14 @@ def spawn(mobs, map_):
...
-                spot_found = (not terrain.is_solid(map_[mx][my]) and
-                              not terrain.is_solid(map_[mx][my - 1]) and
-                              terrain.is_solid(map_[mx][my + 1]))
+                feet = map_[mx][my]
+                head = map_[mx][my - 1]
+                floor = map_[mx][my + 1]
+                spot_found = (not terrain.is_solid(feet) and
+                              not terrain.is_solid(head) and
+                              terrain.is_solid(floor) and
+                              not get_light_level(mx, my) and
```
Comment (olls) [inline mobs.py]: This review thing is weird...

**Comment by grit96** [comment]
You can't approve it yet...

**Comment by olls** [comment]
I didn't mean to start a review

**Comment by olls** [comment]
We will have to generate a larger lighting buffer on the server if we want to keep the same spawning range as we currently have. To support this better, and to prevent having to pass offsets everywhere, we should probably store the coordinates of the lighting buffer inside it.

**Comment by olls** [comment]
I think the lighting map will have to be generated in the LocalInterfaces, this means clients can generate one for the size of the window, and a server generates one the size of the mob spawn range.

**Comment by olls** [comment]
Currently the mobs can spawn anywhere in the server's loaded map, we can't generate a lighting buffer for the entire map as it is too big and not continuous. To fix this we could only spawn mobs within a range of each player and generate lighting buffers for each one, one at a time. This might be too slow though, maybe we should put mob spawning into a background thread...

**Comment by olls** [comment]
Mob spawning is correctly working with the lighting now. The only other thing we might want to do is tweak the spawning conditions a bit.

**Pull request merged**